arrow.go

```diff
@@ -91,6 +91,75 @@ func NewArrowFromConn(driverConn driver.Conn) (*Arrow, error) {
 	return &Arrow{c: dbConn}, nil
 }
 
+// arrowStreamReader implements array.RecordReader for streaming DuckDB results.
+type arrowStreamReader struct {
+	ctx        context.Context
+	stmt       *Stmt
+	res        *C.duckdb_arrow
+	schema     *arrow.Schema
+	rowCount   uint64
+	readCount  uint64
+	currentRec arrow.Record
+	err        error
+}
+
+func (r *arrowStreamReader) Retain() {}
+
+func (r *arrowStreamReader) Release() {
+	if r.currentRec != nil {
+		r.currentRec.Release()
```

Comment:
If this is a `release` callback for the ArrowArray, it should check whether `release` is null before calling it
> A released structure is indicated by setting its release callback to NULL. Before reading and interpreting a structure’s data, consumers SHOULD check for a NULL release callback and treat it accordingly (probably by erroring out).